hudi-common/src/main/java/org/apache/hudi/common/table/log/block/HoodieAvroDataBlock.java

```diff
@@ -192,16 +198,25 @@ public boolean hasNext() {
     }
 
     @Override
-    public IndexedRecord next() {
+    public HoodieRecord next() {
       try {
         int recordLength = this.dis.readInt();
         BinaryDecoder decoder = DecoderFactory.get().binaryDecoder(this.content, this.dis.getNumberOfBytesRead(),
                 recordLength, this.decoderCache.get());
         this.decoderCache.set(decoder);
-        IndexedRecord record = this.reader.read(null, decoder);
+        IndexedRecord indexedRecord = this.reader.read(null, decoder);
         this.dis.skipBytes(recordLength);
         this.readRecords++;
-        return record;
+        if (recordType == HoodieRecordType.SPARK) {
```

Comment:
We should hide this w/in the reader

Reply:
Do you mean that we should move the logic of reading avro data into HoodieAvroReader and HoodieSparkAvroReader?  And we create these readers from HoodieFileReaderFactory. Just like usage of FileReader in HoodieParquetDataBlock.

Reply:
Correct